src/common/errorBoundary/index.ts

```diff
@@ -0,0 +1,31 @@
+// Custom Error class to extend properties to error object
+export class i18nifyError extends Error {
+  timestamp: Date;
+  constructor(message) {
+    super(message);
+    this.name = "i18nify Error";
+    this.timestamp = new Date();
+    // more params like type of error/severity can be added in future for better debugging.
+  }
+}
+
+/**
+ * withErrorBoundary is a higher order function that takes function as parameter and wraps it in try/catch block.
+ * It appends additional attributes and serves as a centralized error-handling service.
+ * Usage =>
+ * const wrappedUtilityFn = withErrorBoundary(utilityFn)
+ *
+ * @param fn utility that is wrapped in error boundary
+ * @returns {Function} returns the function wrapped in try/catch block
+ */
+export const withErrorBoundary = (fn: Function): Function => {
+  return function (...rest) {
```

Comment:
suggestions: use the arrow function to `not to` have context of current function

Reply:
arrow function can't be used here as we have to preserve the value of `this` passed from the consumer. By Using arrow function, it will simply take the global scope which won't bring any benefit. 

For example, if in future someone does this... the object {b:1} should be logged in test function not the window object. 
```
function test(a) {
  console.log(this)
  console.log(a)
}

const wrappedTest = withErrorBoundary(test)

wrappedTest.call({b: 1}, 2)
```